### ISSUE TITLE
Forbid to use `while False` and equivalents

### ISSUE THREAD

**Issue by sobolevn** [wemake-services/wemake-python-styleguide#725]
# Rule request

## Thesis

Currently code:

```python
while False:
    ...
```

Does not a violation on `while False`. But, that's 100% incorrect. This code will never be executed.

We need to forbid: `False`, `0`, `''`, `[]`, `()`, `{}`, `None` with `while` condition.

## Reasoning

This rule is related to `WrongLoopDefinitionVisitor` and `WrongLoopIterTypeViolation`

**Comment by kxepal** [comment]
Can rule be a bit more general, like "Forbid unreachable code"? The same case applies as well for `if`, partially to `for` and allows to catch more things like code at the same block after return / raise etc. pylint does that good.

**Comment by sobolevn** [comment]
This rule already exists. We already know about `if`, `for`, and unreachable blocks. The issue is just in `while`. We have missed it.

**Comment by kxepal** [comment]
Ah, good then.
